src/arch/arm/32/traps.S

```diff
@@ -74,6 +74,12 @@ BEGIN_FUNC(arm_swi_syscall)
 #ifdef CONFIG_FASTPATH
     cmp r7, #SYSCALL_CALL
     beq c_handle_fastpath_call
+#ifdef CONFIG_KERNEL_MCS
+#ifdef CONFIG_SIGNAL_FASTPATH
```

Comment:
if `CONFIG_SIGNAL_FASTPATH` is checked in cmake to imply `CONFIG_KERNEL_MCS`, you don't need to test for both. The fewer ifdefs the better.

Reply:
Makes sense. Currently, it appears that `SIGNAL_FASTPATH` is also dependent on `CONFIG_FASTPATH`, as this section will never be reached if `CONFIG_FASTPATH` isn't set. Do you think this is appropriate, and if so I should add something similar in the cmake config for the signal fastpath which checks `KernelFastpath`, right?

Reply:
In the code you need to check for the most specific one under which you want it to run. You can have `CONFIG_FASTPATH` and  `not  CONFIG_SIGNAL_FASTPATH`, but you can't have `CONFIG_FASTPATH` on false and `CONFIG_SIGNALFASTPATH` on true. So the one to leave out here would be `MCS`, I think.